AnnService/src/Core/VectorIndex.cpp

```diff
@@ -294,6 +294,42 @@ VectorIndex::DeleteIndex(ByteArray p_meta) {
 }
 
 
+ErrorCode
+VectorIndex::MergeIndex(const char* p_indexFilePath)
+{
+    std::string folderPath(p_indexFilePath);
+    if (!folderPath.empty() && *(folderPath.rbegin()) != FolderSep) folderPath += FolderSep;
+    Helper::IniReader iniReader;
+    if (ErrorCode::Success != iniReader.LoadIniFile(folderPath + "indexloader.ini")) return ErrorCode::FailedOpenFile;
+
+    std::shared_ptr<VectorIndex> addIndex = CreateInstance(iniReader.GetParameter("Index", "IndexAlgoType", IndexAlgoType::Undefined),
+        iniReader.GetParameter("Index", "ValueType", VectorValueType::Undefined));
+    addIndex->LoadConfig(iniReader);
+    addIndex->LoadIndexData(folderPath);
+
+    std::shared_ptr<MetadataSet> pMetadata;
+    if (iniReader.DoesSectionExist("MetaData"))
```

Comment:
still, you need handle ELSE case

Reply:
if MetaData section does not exist, we do not need to load the pMetadata which will be nullptr.